solutions/usaco-894.mdx

```diff
@@ -90,6 +127,61 @@ valid assignment of grass types in the $\texttt{ans}$ array.
 
 <LanguageSection>
 
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+
+// USACO-style file input
+void setIO(string name = ""){ // FastIO see General -> Fast Input and Output
+	ios_base::sync_with_stdio(0); cin.tie(0);
+	if(name.size()){
+		freopen((name+".in").c_str(), "r", stdin);
+		freopen((name+".out").c_str(), "w", stdout);
+	}
+}
+
+bool visited[10001];
+int ans[10001];
+vector<int> adjList[10001];
+
+void dfs(int self, int parent){
+	int len = adjList[self].size();
+	int g = 1;
+	for (int i : adjList[self]) {
+		if (i != parent) {
+			while (g == ans[self] || g == ans[parent]) g++;
+				ans[i] = g;
+				dfs(i, self);
+				g++;
+			}
+		}
+}
+
+int main() {
+	setIO("planting");
+
+	int n; cin >> n;
+	int u, v;
+	for(int i=1;i<n;i++){
+		cin>>u>>v;
+		adjList[u].push_back(v);
+		adjList[v].push_back(u);
+	}
+	dfs(1, 1);
+	int maxAns = 0;
+	for (int curAns : ans) {
+		maxAns = max(maxAns, curAns);
+	}
+	cout << maxAns << "\n";
+	return 0;
```

Comment:
```suggestion
```

Reply:
Little confused here, why not just return 0; when function return type is int.

Reply:
the main function returns 0 by default https://stackoverflow.com/questions/4138649/why-is-return-0-optional